packages/babel-parser/test/fixtures/flow/regression/arrow-in-ternary-with-param-type-and-return-type-like/input.js

```diff
@@ -0,0 +1,2 @@
+test
+  ? (x: T): U => y
```

Comment:
`U => y` must be parenthesized, otherwise flow [could not](https://flow.org/try/#0C4UwzsBQAE0PzQBQA8Bc0AqBKdBVaAvAHzQCeQA) parse it.

Reply:
It is a known bug with flow, because it also rejects this valid JS code:
```js
test ? (x) : I => y
```

https://github.com/facebook/flow/issues/2006

Reply:
For context, here is one of my first PRs :stuck_out_tongue: 
https://github.com/babel/babylon/pull/573